engine/Register.ts

```diff
@@ -1,7 +1,67 @@
+import { PresentationSystemGroup, defineComponent, defineQuery, defineSystem } from '@etherealengine/ecs'
 import { ComponentShelfCategoriesState } from '@etherealengine/editor/src/components/element/ElementList'
+import { ComponentEditorsState } from '@etherealengine/editor/src/functions/ComponentEditors'
 import { GrabbableComponent } from '@etherealengine/engine/src/interaction/components/GrabbableComponent'
-import { getMutableState } from '@etherealengine/hyperflux'
-import './benchmarks/BenchmarkOrchestration'
-import './benchmarks/Profiling'
+import { SceneState } from '@etherealengine/engine/src/scene/Scene'
+import { getMutableState, useHookstate, useMutableState } from '@etherealengine/hyperflux'
+import { useEffect } from 'react'
+import { BenchmarkComponentNodeEditor, ProfilingComponentNodeEditor } from './benchmarks/BenchmarkNodeEditors'
+
+export const ProfilingComponent = defineComponent({
+  name: 'eepro.eetest.ProfilingComponent',
+  jsonID: 'eepro.eetest.ProfilingComponent'
+})
+
+export const BenchmarkComponent = defineComponent({
+  name: 'eepro.eetest.BenchmarkComponent',
+  jsonID: 'eepro.eetest.BenchmarkComponent'
+})
 
 getMutableState(ComponentShelfCategoriesState).Interaction.merge([GrabbableComponent])
+getMutableState(ComponentShelfCategoriesState).merge({
+  Debug: [ProfilingComponent, BenchmarkComponent]
+})
+
+getMutableState(ComponentEditorsState).merge({
+  [ProfilingComponent.name]: ProfilingComponentNodeEditor,
+  [BenchmarkComponent.name]: BenchmarkComponentNodeEditor
+})
+
+const ProfilingComponentQuery = defineQuery([ProfilingComponent])
+const BenchmarkComponentQuery = defineQuery([BenchmarkComponent])
+
+const reactor = () => {
+  const importedProfiling = useHookstate(false)
+  const importedBenchmark = useHookstate(false)
+  const sceneState = useMutableState(SceneState)
+
+  useEffect(() => {
+    if (!sceneState.sceneLoaded.value) return
```

Comment:
we should avoid reliance on sceneLoaded if possible. we can simplify this into useQuery calls to decouple from needing a scene to be loaded at all